Apps/Inc/Tasks.h

```diff
@@ -134,36 +128,60 @@ typedef enum{
     OS_MAIN_LOC = 0x200,
     OS_CANDRIVER_LOC = 0x400,
     OS_MOTOR_CONNECTION_LOC = 0x800,
-    OS_DISPLAY_LOC = 0x1000
+    OS_DISPLAY_LOC = 0x1000,
+    OS_TASKS_LOC = 0x2000,
 } os_error_loc_t;
 
-/**
- * Fault Enum
- * 
- * Different fault states that need to be handled by the FaultState task
- */
-typedef enum{
-    FAULT_NONE = 0x00,      // No fault
-    FAULT_OS = 0x01,         // for OS faults
-    FAULT_UNREACH = 0x02,    // for unreachable conditions
-    FAULT_TRITIUM = 0x04,      // for errors sent from the tritium
-    FAULT_READBPS = 0x08,    // for unsuccessfully reading from BPS CAN
-    FAULT_DISPLAY = 0x10,    // for display faults
-    FAULT_BPS = 0x20,       // for if BPS trips
-} fault_bitmap_t;
-
 /**
  * Error variables
  */
-extern fault_bitmap_t FaultBitmap;
 extern os_error_loc_t OSErrLocBitmap;
 
+// Store error codes that are set in task error assertion functions
+extern error_code_t Error_ReadTritium; 
+extern error_code_t Error_ReadCarCAN;
+extern error_code_t Error_UpdateDisplay;
+
+/**
+ * Error-handling option enums
+*/
+
+// Scheduler lock parameter option for asserting a task error
+typedef enum {
+    OPT_LOCK_SCHED = false,
+    OPT_NO_LOCK_SCHED = true
+} error_lock_sched_opt_t;
+
+// Recoverable/nonrecoverable parameter option for asserting a task error
+typedef enum {
+    OPT_RECOV,
+    OPT_NONRECOV
+} error_recov_opt_t;
+
+/**
+ * @brief For use in error handling: turns off array, array precharge, and motor contactor
+ * and turns on additional brakelight to signal that a critical error happened.
+*/
+void arrayMotorKill();
+
+/**
+ * @brief Assert a task error by setting the location variable and optionally locking the scheduler, 
+ * displaying a fault screen (if nonrecoverable), jumping to a callback function, and entering an infinite loop. 
+ * Called by task-specific error-assertion functions that are also responsible for setting the error variable.
+ * @param errorLoc the task from which the error originated. Note: should be taken out when last task pointer is integrated
+ * @param faultCode the enum for the specific error that happened
+ * @param errorCallback a callback function to a handler for that specific error, 
+ * @param schedLock whether or not to lock the scheduler to ensure the error is handled immediately
+ * @param nonrecoverable whether or not to kill the motor, display the fault screen, and enter an infinite while loop
+*/
+void assertTaskError(os_error_loc_t errorLoc, uint8_t faultCode, callback_t errorCallback, error_lock_sched_opt_t lockSched, error_recov_opt_t nonrecoverable);
+
 /**
  * @brief   Assert Error if OS function call fails
  * @param   OS_err_loc Where OS error occured (driver level)
  * @param   err OS Error that occurred
  */
-void _assertOSError(uint16_t OS_err_loc, OS_ERR err); //TODO: This should be changed to enforce only enum usage
+void _assertOSError(uint16_t OS_err_loc, OS_ERR err); 
```

Comment:
Is there a reason this is a uint16_t instead of the typedef'ed type? If you typedef something, consistently use it, or else its not valuable.

Reply:
Gotcha, I can't think of any reason why, so I'll fix that.